### PR TITLE
Life mngr: add Makefile and adjust source

### PR DESCRIPTION
**Pull request by fuyanX**

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: misc/life_mngr/Makefile [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+OUT_DIR ?= $(shell mkdir -p $(T)/build;cd $(T)/build;pwd)
+CC ?= gcc
+
+LIFEMNGR_CFLAGS := -g -O0 -std=gnu11
+LIFEMNGR_CFLAGS += -D_GNU_SOURCE
+LIFEMNGR_CFLAGS += -DNO_OPENSSL
+LIFEMNGR_CFLAGS += -m64
+LIFEMNGR_CFLAGS += -Wall -ffunction-sections
+LIFEMNGR_CFLAGS += -Werror
+LIFEMNGR_CFLAGS += -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=2
```
Comment (binbinwu1) [inline misc/life_mngr/Makefile]: Is it reasonable to specify both O0 and O2 at the same time?

File: misc/life_mngr/life_mngr.c [hunk 2]
```diff
@@ -98,16 +99,19 @@ int main(int argc, char *argv[])
 	do {
 		if (node == NODE_UOS_SERVER) {
 			memset(recvbuf, 0, sizeof(recvbuf));
-			read(fd_uos, recvbuf, sizeof(recvbuf));
+			ret = read(fd_uos, recvbuf, sizeof(recvbuf));
```
Comment (binbinwu1) [inline misc/life_mngr/life_mngr.c]: no need to check the return value?

File: misc/life_mngr/Makefile [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+endif
+endif
+endif
+
+LIFEMNGR_LDFLAGS := -Wl,-z,noexecstack
+LIFEMNGR_LDFLAGS += -Wl,-z,relro,-z,now
+LIFEMNGR_LDFLAGS += -pie
+LIFEMNGR_LDFLAGS += $(LDFLAGS)
+
+all:
```
Comment (binbinwu1) [inline misc/life_mngr/Makefile]: need target install in this Makefile?

Comment (fuyanX) [reply misc/life_mngr/Makefile]: lifemngr is running on UOS, not on SOS.
So it no need to install.

File: misc/life_mngr/life_mngr.c (hunk 2 above)
Comment (fuyanX) [reply misc/life_mngr/life_mngr.c]: This "ret" is just for clear warning.
"ret" is no need to check, as the "recvbuf" is checked.

File: misc/life_mngr/Makefile (hunk 1 above)
Comment (fuyanX) [reply misc/life_mngr/Makefile]: This Makefile refers to other tool's Makefile.
In order to keep them uniformity.

Comment (binbinwu1) [reply misc/life_mngr/Makefile]: But it is not right to do so, for this patch,  just keep O2? need other patch to correct the other Makefile

Comment (fuyanX) [reply misc/life_mngr/Makefile]: OK. I will correct it.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by binbinwu1** [comment]
@fuyanX, could you edit the title of this PR according to the change it made? The current title is not clear to me.

**Pull request merged**